packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -472,44 +494,25 @@ contract Deploy is Deployer {
         require(delayedVetoable.vetoer() == cfg.superchainConfigVetoer());
         vm.stopPrank();
 
-        save(contractName, address(delayedVetoable));
-        console.log("%s deployed at %s", contractName, address(delayedVetoable));
+        save(_contractName, address(delayedVetoable));
+        console.log("%s deployed at %s", _contractName, address(delayedVetoable));
     }
 
-    /// @notice Transfer ownership of the SuperchainConfigProxy to the DelayedVetoableForSuperchain contract
-    function transferSuperchainConfigProxyOwnership() public broadcast {
-        address superchainConfigProxy = mustGetAddress("SuperchainConfigProxy");
-        address delayedVetoableForSuperchain = mustGetAddress("DelayedVetoableForSuperchain");
-        _callViaSafe({
-            _safe: Safe(mustGetAddress("SuperchainConfigInitiatorSafe")),
-            _target: superchainConfigProxy,
-            _data: abi.encodeCall(Proxy.changeAdmin, (delayedVetoableForSuperchain))
-        });
-        console.log("SuperchainConfigProxy ownership transferred to %s", delayedVetoableForSuperchain);
-    }
-
-    /// @notice Transfer ownership of the ProxyAdmin to the DelayedVetoableForOpChain contract
-    function transferProxyAdminOwnership() public broadcast {
-        address proxyAdmin = mustGetAddress("ProxyAdmin");
-        address delayedVetoableForOpChain = mustGetAddress("DelayedVetoableForOpChain");
-        _callViaSafe({
-            _safe: Safe(mustGetAddress("SuperchainConfigInitiatorSafe")),
-            _target: proxyAdmin,
-            _data: abi.encodeCall(Ownable.transferOwnership, (delayedVetoableForOpChain))
-        });
-        console.log("ProxyAdmin ownership transferred to %s", delayedVetoableForOpChain);
-    }
-
-    /// @notice Transfer ownership of the ProtocolVersions to the DelayedVetoableForProtocolVersions contract
-    function transferProtocolVersionsProxyOwnership() public broadcast {
-        address protocolVersionsProxy = mustGetAddress("ProtocolVersionsProxy");
-        address delayedVetoableForProtocolVersions = mustGetAddress("DelayedVetoableForProtocolVersions");
+    /// @notice Transfers control of a contract controlled by the initiator safe
+    function transferControl(
+        string memory _controlled,
+        string memory _newController,
+        bytes4 _transferSelector
+    )
+        public
+        broadcast
+    {
         _callViaSafe({
             _safe: Safe(mustGetAddress("SuperchainConfigInitiatorSafe")),
-            _target: protocolVersionsProxy,
-            _data: abi.encodeCall(Proxy.changeAdmin, (delayedVetoableForProtocolVersions))
+            _target: mustGetAddress(_controlled),
+            _data: abi.encodeWithSelector(_transferSelector, mustGetAddress(_newController))
```

Comment:
This has an assumption that the function will always accept a single address arg

Reply:
Yes, but I think it's fine for now. AFAIK in the whole codebase we don't have any ownership transfer like functions which take two args. 

Supporting an arbitrary number of arguments is possible using some variant on `abi.encodeX()` or just passing in the calldata, but that feels to me like it walks back the value of abstracting this operation into a single reusable method.

Reply:
Going to resolve, please feel free to reopen if this is important to you.